### DISCUSSION TITLE
Workflow can't read a private Flutter package hosted on Github

### DISCUSSION THREAD

**Discussion by anarenteriareyes** [codemagic-ci-cd/codemagic-docs#2331] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Hi, I have a problem reading a private flutter package hosted in a private Github repository from a Codemagic workflow.
I have followed Codemagic documentation on "[Accessing private dependencies and Git submodules](https://docs.codemagic.io/knowledge-git/access-private-git-submodules)" but my build fails to complete the "flutter pub get" command and I get the error message "remote: Repository not found." which I think may be related to SSH. 

I did the following steps:
- Generate an ssh key
- Add the public ssh key to my private GitHub repository as a deployment key (I also tested adding the ssh key in my account settings).
- Add the private ssh key to my app environment variables in Codemagic

Any help would be appreciated, thanks!

### Expected results

Be able to read a private flutter package in GitHub from the Codemagic workflow

### Actual results

The workflow fails because it cannot complete the "flutter pub get" command to download the flutter packages including the private flutter package hosted on Github.

**Comment by himesh-cm** [comment]
Hi @anarenteriareyes, 
Can you confirm if the environment variable name ends in `_SSH_KEY`?
Please share the build ID with me to investigate the issue further.
You can find it under your build details page's 'ID' section. ﻿Navigate to the 'Builds' section on the left-hand menu, select the build you want to view, and click the copy icon.

**Comment by himesh-cm** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
